scripts/migration/lib.sh

```diff
@@ -281,6 +282,14 @@ check_version() {
     fi
   fi
 
+  cluster_version="$(get_apps_version "${1}" >/dev/null 2>&1 || true)"
+  if [ -n "${cluster_version}" ]; then
+    log_info "Currently running ${cluster_version} according to cluster"
+    if [ "${common_override}" != "null" ] && [ "${cluster_version%.*}" != "${common_override}" ]; then
+      log_warn "Version mismatch, cluster ${cluster_version}, config ${common_override}"
+    fi
+  fi
+
```

Comment:
Question: Could you elaborate on why we have this block? If I understand it correctly then it will warn if there is a version override in the common config and that is not the same as the applied version. But it will not exit and is not part of a prompt, so it would just be informational when looking at what is happening.

Could it perhaps make sense to merge this check into the prompt above it? Then you would still get this information, but it can be used as more context to decide if you want to abort or not. What do you think of that?

Reply:
Hm, probably missed that it's the _override_ config it checks against. The intent was probably to check the pre-upgrade config version (`defaults/common-config.yaml → .global.ck8sVersion`) against the in-cluster version (if available). E.g. if you try to upgrade to `v0.43` and your config says `v0.42` but your cluster says `v0.41` or similar. This gets complicated and hard to think about since the config is changed by `ck8s init`.

Maybe just drop this, or what do you think is the way forward?

Reply:
I'm not sure what the best way forward is.
Merging it with the block above might still be best, so that you get a chance to abort. Because I think that there is some value to having this check.

Reply:
I moved the block above the abort check but removed the mismatch check since it would be comparing the to-be-upgraded-to version recorded by `ck8s init` to the currently running version, which wasn't the intent and it would always mismatch unless try to migrate a second time.